### DISCUSSION TITLE
Separate renovate from docker images

### DISCUSSION THREAD

**Discussion by ekhaydarov** [renovatebot/renovate#20026] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.112.2

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

currently the output is like this. renovate gets grouped at the end. 
```
chore(docker): update docker images (alpine, artifacts-we1.farfetch.net/farfetch-data-docker-stable-we1/farfetch/data-infrastructure-argocd, artifacts-we1.farfetch.net/farfetch-data-docker-stable-we1/farfetch/konnekt-monitoring-airflowexporter, docker.io/grafana/grafana, ghcr.io/outsideris/citizen, palantirtechnologies/policy-bot, postgres, rclone/rclone, renovate/renovate)
```
i would like to split out renovate to automerge itself. so i have tried to put it in package rules as seen below. i tried it above and below the docker grouping but it doesnt get picked up. what am i missing?

### What you tried so far.
```
packageRules: [
  { 
      semanticCommitScope: "self",
      matchDepNames:[
        "renovate/renovate"
      ],
    },
    {
      groupName: "docker images",
      semanticCommitScope: "docker",
      matchDatasources: ["docker"],
      separateMajorMinor: "false"
    },
]
```
### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by viceice** [comment, marked as answer]
use exclude package for second rule. renovate applies all matching rules in order

**Reply by ekhaydarov** [reply]
if it applies matching rules in order shouldnt the below get me what i want? if renovate is first and then docker takes renovate into its own grouping. would the below do the same and renovate take it out of the docker grouping. that doesnt seem to happen. 


```
packageRules: [
    {
      groupName: "docker images",
      semanticCommitScope: "docker",
      matchDatasources: ["docker"],
      separateMajorMinor: "false"
    },
    { 
      semanticCommitScope: "self",
      matchDepNames:[
        "renovate/renovate"
      ],
    },
]
```

I am merely confused about the rules, the [config options](https://docs.renovatebot.com/configuration-options) while explaining what each individual one does clearly, do not explain how they all work together quite enough. are there any more in depth config docs or some more nuanced examples?

**Reply by viceice** [reply]
no, both rules match so both are applied

**Reply by ekhaydarov** [reply]
if i add `excludeDepName` then i get a separate renovate. without that it seems both dont match to create a pr? there must be a further pr deduplication that removes renovate from standalone and gives it to docker. i assumed the ordering would determine how deduplication goes, but it has no effect as previously stated
